### ISSUE TITLE
Completion of error handling

### ISSUE THREAD

**Issue by elfring** [100/Cranium#4]
Would you like to add more error handling for return values from functions like the following?
- [malloc](http://pubs.opengroup.org/onlinepubs/9699919799/functions/malloc.html "Memory allocation") ⇒ [createLayer](https://github.com/100/Cranium/blob/4c1d028a4f9e897a3f709a6b92cae414eba6ee59/src/layer.h#L56)
- [printf](http://pubs.opengroup.org/onlinepubs/9699919799/functions/printf.html "Print formatted output") ⇒ [printMatrix](https://github.com/100/Cranium/blob/7d5c219c9bd62b4a92bda712c374cc786c9be0ed/src/matrix.h#L137)

**Comment by 100** [comment]
Yeah I think that's definitely something useful, I'm just not sure what the best way to handle this would be. E.g. should it exit upon receiving a null matrix, network, etc. or should we add some other mechanism for checking status built into the library

**Comment by elfring** [comment]
I suggest to avoid [ignorance of return values](https://cwe.mitre.org/data/definitions/252.html "Checking more return values") a bit more.
Would you like to detect every error situation as early as possible?

**Comment by 100** [comment]
Yes, it's probably best to detect these within functions as they occur and subsequently exit, but allow functions to return null pointers if they are user-exposed (networks, matrices)

**Comment by elfring** [comment]
* Are you interested to apply [aspect-oriented software development](http://aspectc.org/ "Software development with AspectC++?")?
* How do you think about to encapsulate error detection and corresponding exception handling as [a reusable aspect](https://en.wikipedia.org/wiki/Aspect_%28computer_programming%29 "Combination of a pointcut with advice") in your software?
